homeassistant/components/xbox/browse_media.py

```diff
@@ -0,0 +1,182 @@
+    MEDIA_CLASS_GAME,
+    MEDIA_TYPE_APP,
+    MEDIA_TYPE_GAME,
+)
+
+TYPE_MAP = {
+    "App": {
+        "type": MEDIA_TYPE_APP,
+        "class": MEDIA_CLASS_APP,
+    },
+    "Game": {
+        "type": MEDIA_TYPE_GAME,
+        "class": MEDIA_CLASS_GAME,
+    },
+}
+
+
+async def build_item_response(
+    client: XboxLiveClient,
+    device_id: str,
+    tv_configured: bool,
+    media_content_type: str,
+    media_content_id: str,
+) -> Optional[BrowseMedia]:
+    """Create response payload for the provided media query."""
+    apps: InstalledPackagesList = await client.smartglass.get_installed_apps(device_id)
+
+    if media_content_type in [None, "library"]:
+        library_info = BrowseMedia(
+            media_class=MEDIA_CLASS_DIRECTORY,
+            media_content_id="library",
+            media_content_type="library",
+            title="Installed Applications",
+            can_play=False,
+            can_expand=True,
+            children=[],
+        )
+
+        # Add Home
+        id_type = AlternateIdType.LEGACY_XBOX_PRODUCT_ID
+        home_catalog: CatalogResponse = (
+            await client.catalog.get_product_from_alternate_id(
+                HOME_APP_IDS[id_type], id_type
+            )
+        )
+        home_thumb = _find_media_image(
+            home_catalog.products[0].localized_properties[0].images
+        )
+        library_info.children.append(
+            BrowseMedia(
+                media_class=MEDIA_CLASS_APP,
+                media_content_id="Home",
+                media_content_type=MEDIA_TYPE_APP,
+                title="Home",
+                can_play=True,
+                can_expand=False,
+                thumbnail=home_thumb.uri,
+            )
+        )
+
+        # Add TV if configured
+        if tv_configured:
+            tv_catalog: CatalogResponse = (
+                await client.catalog.get_product_from_alternate_id(
+                    SYSTEM_PFN_ID_MAP["Microsoft.Xbox.LiveTV_8wekyb3d8bbwe"][id_type],
+                    id_type,
+                )
+            )
+            tv_thumb = _find_media_image(
+                tv_catalog.products[0].localized_properties[0].images
+            )
+            library_info.children.append(
+                BrowseMedia(
+                    media_class=MEDIA_CLASS_APP,
+                    media_content_id="TV",
+                    media_content_type=MEDIA_TYPE_APP,
+                    title="Live TV",
+                    can_play=True,
+                    can_expand=False,
+                    thumbnail=tv_thumb.uri,
+                )
+            )
+
+        content_types = sorted(
+            {
+                app.content_type
+                for app in apps.result
+                if app.content_type in TYPE_MAP.keys()
+            }
+        )
+        for c_type in content_types:
+            library_info.children.append(
+                BrowseMedia(
+                    media_class=TYPE_MAP[c_type]["class"],
```

Comment:
This seems to be a directory of apps or games, and should have media class `MEDIA_CLASS_DIRECTORY`.